Platform/CoffeelakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -570,15 +570,12 @@ GetPlatformPowerState (
     }
   }
 
-  if ((MmioRead8 (PCH_PWRM_BASE_ADDRESS + R_PMC_PWRM_GEN_PMCON_B) & B_PMC_PWRM_GEN_PMCON_B_RTC_PWR_STS) != 0) {
-    BootMode = BOOT_WITH_FULL_CONFIGURATION;
-
-    ///
-    /// Clear Wake Status and Sleep Type
-    ///
-    IoWrite16 (ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_STS, B_ACPI_IO_PM1_STS_WAK);
-    IoAndThenOr16 (ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_CNT, (UINT16) ~B_ACPI_IO_PM1_CNT_SLP_TYP, V_ACPI_IO_PM1_CNT_S0);
-  }
+  ///
+  /// Clear Wake Status and Sleep Type
+  /// Also clear the PWRBTN_EN, it causes SMI# otherwise
+  ///
+  IoWrite32 (ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_STS, ((UINT32)~B_ACPI_IO_PM1_EN_PWRBTN_EN & R_ACPI_IO_PM1_EN_MASK) | B_ACPI_IO_PM1_STS_WAK );
+  IoAndThenOr16 (ACPI_BASE_ADDRESS + R_ACPI_IO_PM1_CNT, (UINT16) ~B_ACPI_IO_PM1_CNT_SLP_TYP, V_ACPI_IO_PM1_CNT_S0);
```

Comment:
Why do we clear the sleep type unconditionally?   And why not check 
B_PMC_PWRM_GEN_PMCON_B_RTC_PWR_STS any more ?

Reply:
SInce we captured the boot mode already, we can clear the wake status & sleep type here. RTC_PWR_STS check  was meant for only if there is an issue with RTC battery. We can add that check after this code and set to Full configuration BootMode. But clearing SleepType and WakeStatus can be done anyways, right?

Reply:
SleepStatus can be cleared,  however, SleepType should not be cleared.  I remember OS also needs to use this information.  
The original logic seems to be reasonable for me.   If you just need to clear PWRBTN_EN, you should just do that without changing other logic.